core/domain/fs_domain.py

```diff
@@ -108,96 +107,105 @@ def version(self):
         return self._version
 
 
-class ExplorationFileSystem(object):
-    """A datastore-backed read-write file system for a single exploration.
+class EntityFileSystem(object):
+    """A datastore-backed read-write file system for a single entity.
 
-    The conceptual intention is for each exploration to have its own asset
-    folder. An asset has no meaning outside its exploration, so the assets in
-    these asset folders should therefore not be edited directly. They should
-    only be modified as side-effects of some other operation (such as adding an
-    image to an exploration).
-
-    The content of an exploration should include a reference to the asset
-    together with the version number of the asset. This allows the
-    exploration to refer to asset versions.
+    The conceptual intention is for each entity type to have its own parent
+    folder. In this, each individual object will have its own folder with the
+    corresponding ID as the folder name. These folders will then have the assets
+    folder inside which stores images, audio etc (example path:
+    story/story_id/assets/).An asset has no meaning outside its entity, so the
+    assets in these asset folders should therefore not be edited directly. They
+    should only be modified as side-effects of some other operation (such as
+    adding an image to an entity like topics, stories or explorations).
 
     In general, assets should be retrieved only within the context of the
-    exploration that contains them, and should not be retrieved outside this
+    entity that contains them, and should not be retrieved outside this
     context.
 
     Args:
-        exploration_id: str. The id of the exploration.
+        entity_name: str. The name of the entity (eg: exploration, topic etc).
+        entity_id: str. The ID of the corresponding entity.
     """
 
     _DEFAULT_VERSION_NUMBER = 1
 
-    def __init__(self, exploration_id):
-        """Constructs a ExplorationFileSystem object.
+    def __init__(self, entity_name, entity_id):
+        """Constructs a EntityFileSystem object.
 
         Args:
-            exploration_id: str. The id of the exploration.
+        entity_name: str. The name of the entity (eg: exploration, topic etc).
+        entity_id: str or None. The ID of the corresponding entity. None is to
+            accommodate tests in some files and some older exp jobs where a
+            single exp_id is used to initialize class.
         """
-        self._exploration_id = exploration_id
+        if entity_id is None:
```

Comment:
We should validate that entity_name is valid and that entity_id is in the right format (non-empty string).

Reply:
Done